### PR TITLE
Parser for decision definitions XML

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Related to https://github.com/open-formulieren/open-forms/issues/3855

### CODE REVIEW

File: django_camunda/dmn/utils.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 from datetime import datetime
-from typing import Tuple
+from typing import List, Tuple
```
Comment (SilviaAmAm) [inline django_camunda/dmn/utils.py:2]: This library supports python 3.8

Comment (sergei-maertens) [reply django_camunda/dmn/utils.py:2]: I'm okay with dropping support for anything older than 3.10, let's tackle that at another time :)

File: django_camunda/dmn/utils.py [hunk 2]
```diff
@@ -63,3 +65,57 @@ def parse_dmn(xml: str) -> IntrospectionResult:
...
     outputs = dict((process_output_var(output_var) for output_var in output_vars))
     return IntrospectionResult(inputs=inputs, output=outputs)
+
+
+def extract_decision_definition_inputs(
+    xml: _Element, definition_id: str
+) -> List[DMNInputParameter]:
+    decision_table = xml.find(
+        f".//decision[@id='{definition_id}']", namespaces=xml.nsmap
+    )
```
Comment (Viicos) [inline django_camunda/dmn/utils.py:75]: Not sure if the actual xml tree contains sensitive data, but might be better to escape the xpath query:

```python
xml.xpath("//decision[@id=$definition_id]", definition_id=definition_id)
```

File: tests/dmn/data/xml.py
Comment (sergei-maertens) [inline tests/dmn/data/xml.py:1]: Let's just create a `drd.xml` file instead and load that in the test:

```py
from pathlib import Path

TEST_DRD_XML = Path("./drd.xml").read_text()  # or read_bytes()
```
I'm quite fond of syntax highlighting, especially for a verbose format like XML :)

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: django_camunda/dmn/utils.py [hunk 3]
```diff
@@ -63,3 +65,57 @@ def parse_dmn(xml: str) -> IntrospectionResult:
...
+    output_vars = [
+        DMNOutputParameter(
+            id=element.attrib["id"],
+            label=element.attrib["label"],
+            name=element.attrib["name"],
+            type_ref=element.attrib["typeRef"],
+        )
+        for element in output_elements
+    ]
+    return output_vars
```
Comment (Viicos) [inline django_camunda/dmn/utils.py:121]:
```suggestion
    return [
        DMNOutputParameter(
            id=element.attrib["id"],
            label=element.attrib["label"],
            name=element.attrib["name"],
            type_ref=element.attrib["typeRef"],
        )
        for element in output_elements
    ]
```
Really small performance improvement I guess

File: tests/dmn/data/xml.py
Comment (Viicos) [reply tests/dmn/data/xml.py:1]: Tip for Pycharm: https://www.jetbrains.com/help/pycharm/using-language-injections.html#use-language-injection-comments

(If you ever have to inline in Python).

This is not a standard and other IDEs won't pick it up so it's mainly useful during development

File: django_camunda/dmn/utils.py (hunk 2 above)
Comment (SilviaAmAm) [reply django_camunda/dmn/utils.py]: Good point :thinking: 
Annoying that XPath cannot deal with the default namespace :(

File: tests/dmn/test_parser.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+from pathlib import Path
+
+from django_camunda.dmn.parser import Parser
+
+TEST_DRD_XML = Path(Path(__file__).parent / "data" / "drd.xml").read_bytes()
```
Comment (sergei-maertens) [inline tests/dmn/test_parser.py:5]:
```suggestion
TEST_DRD_XML = (Path(__file__).parent / "data" / "drd.xml").read_bytes()
```
should be sufficient :) the `/` operator returns a new path instance already

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
While working on #14 I noticed we already have some introspection functionality, please check `django_camunda.dmn.api.introspect_dmn` which also implements the call to retrieve the XML from the camunda API.

I checked that it's unused in Open Forms, so you can make breaking changes in there. I recon that a signature like below is appropriate:

```py
def get_dmn_parser(
    dmn_key: str, 
    *, 
    dmn_id: str = "", 
    client: Optional[Camunda] = None
) -> Parser: ...
```

File: django_camunda/dmn/datastructures.py [hunk 5]
```diff
@@ -10,6 +12,6 @@ class DMNVariable:
 
 
 @dataclass
-class IntrospectionResult:
-    inputs: List[DMNVariable]
-    output: Dict[str, DMNVariable]
+class DMNIntrospectionResult:
+    inputs: List[DMNInputParameter]
+    outputs: List[DMNOutputParameter]
```
Comment (sergei-maertens) [inline django_camunda/dmn/datastructures.py:17]: This can be deleted completely I think.

Comment (SilviaAmAm) [reply django_camunda/dmn/datastructures.py:17]: I left it because I wanted to use it in OF :thinking: Maybe I should move it there?

Comment (SilviaAmAm) [reply django_camunda/dmn/datastructures.py:17]: Or maybe I should add a method to the parser `extract_parameters` that returns this :thinking:

**Review by sergei-maertens — APPROVED** [review]

File: django_camunda/dmn/datastructures.py (hunk 5 above)
Comment (sergei-maertens) [reply django_camunda/dmn/datastructures.py:17]: the latter is nicest IMO :)

**Pull request merged**